server/src/main/java/org/elasticsearch/index/query/SpanMultiTermQueryBuilder.java

```diff
@@ -124,22 +134,61 @@ public static SpanMultiTermQueryBuilder fromXContent(XContentParser parser) thro
     protected Query doToQuery(QueryShardContext context) throws IOException {
         Query subQuery = multiTermQueryBuilder.toQuery(context);
         float boost = AbstractQueryBuilder.DEFAULT_BOOST;
-        if (subQuery instanceof BoostQuery) {
+        if (subQuery instanceof ConstantScoreQuery) {
+            subQuery = ((ConstantScoreQuery) subQuery).getQuery();
+        } else if (subQuery instanceof BoostQuery) {
```

Comment:
should we unwrap recursively to be more resilient to changes to the way queries are built?